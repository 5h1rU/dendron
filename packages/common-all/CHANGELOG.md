# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.11](https://github.com/dendronhq/dendron/compare/v0.3.10...v0.3.11) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-all

## [0.3.10](https://github.com/dendronhq/dendron/compare/v0.1.6...v0.3.10) (2020-07-13)

### Features

- auto add nodes when deleted or created outside of dendron ([8c311bd](https://github.com/dendronhq/dendron/commit/8c311bda948a1d54088c49fd70eb65d24af5d68f))
- better schema suggestions ([03656bc](https://github.com/dendronhq/dendron/commit/03656bc007810457cb6846f0d6adacab4a7fbd3a))
- match namespace schemas ([7a67b8b](https://github.com/dendronhq/dendron/commit/7a67b8b2fb7caa1b97ee6d492d2801782abecdf6))
- show node descriptions ([aca86f2](https://github.com/dendronhq/dendron/commit/aca86f2a5fd6ee481f93553693a098db0e322890))
- surface unknown schemas ([d014965](https://github.com/dendronhq/dendron/commit/d0149652c985c69a4b2607984d578902820077f1))
- updated icons for schemas ([21804eb](https://github.com/dendronhq/dendron/commit/21804eba61c8dd49e499edd5d548d9d601224e8e))

## [0.3.9](https://github.com/dendronhq/dendron/compare/v0.3.8...v0.3.9) (2020-07-09)

### Features

- updated icons for schemas ([b7a2d8a](https://github.com/dendronhq/dendron/commit/b7a2d8aa517cf88d7a93d07cd2ef19305e48d069))

## [0.3.7](https://github.com/dendronhq/dendron/compare/v0.3.6...v0.3.7) (2020-07-08)

### Features

- better schema suggestions ([ad74bc0](https://github.com/dendronhq/dendron/commit/ad74bc009e1544319a49689394ab8d6b684f6578))
- show node descriptions ([e08fce9](https://github.com/dendronhq/dendron/commit/e08fce994153e28fe504b85e6d9bc1f5fdd93e20))

## [0.3.6](https://github.com/dendronhq/dendron/compare/v0.3.5...v0.3.6) (2020-07-07)

### Features

- auto add nodes when deleted or created outside of dendron ([a7e1ac9](https://github.com/dendronhq/dendron/commit/a7e1ac9b8a4f7f0592ab1b9f86a7a40182693a73))

# [0.3.0](https://github.com/dendronhq/dendron/compare/v0.2.20...v0.3.0) (2020-07-05)

### Features

- match namespace schemas ([1a960bf](https://github.com/dendronhq/dendron/commit/1a960bf26f8984e541b3eb118f60bdc09d8250fe))
- surface unknown schemas ([9bf4d0e](https://github.com/dendronhq/dendron/commit/9bf4d0e61cce2f76bddae1f686f29474201466cb))
