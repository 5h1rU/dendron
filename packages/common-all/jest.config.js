const jestConfig = require("../../jest.config");

console.log(jestConfig);
module.exports = {
  ...jestConfig
};
