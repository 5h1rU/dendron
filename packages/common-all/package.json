{
  "name": "@dendronhq/common-all",
  "version": "0.1.6",
  "description": "common-all",
  "license": "MIT",
  "repository": "https://github.com/dendronhq/dendron",
  "author": {
    "name": "Kevin Lin",
    "email": "kevin@thence.io",
    "url": "https://kevinslin.com",
    "organization": true
  },
  "keywords": [
    "common"
  ],
  "files": [
    "lib"
  ],
  "main": "lib/index",
  "types": "lib/index",
  "scripts": {
    "clean": "rimraf lib && rimraf coverage",
    "format": "",
    "lint": "",
    "prepublishOnly": "npm run build",
    "prebuild": "npm run clean && npm run format && npm run lint && echo Using TypeScript && tsc --version",
    "build": "tsc --pretty",
    "test": "jest",
    "coverage": "jest --coverage",
    "watch": "npm run build -- --watch",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "dropbox": "^4.0.30",
    "isomorphic-fetch": "^2.2.1",
    "lodash": "^4.17.15",
    "pino": "^6.3.1",
    "uuid": "^8.1.0",
    "yamljs": "^0.3.0"
  },
  "devDependencies": {
    "@types/jest": "^23.3.3",
    "@types/lodash": "^4.14.152",
    "@types/node": "^10.11.4",
    "@types/pino": "^6.0.1",
    "@types/uuid": "^8.0.0",
    "@types/yamljs": "^0.2.31",
    "coveralls": "^3.0.2",
    "jest": "^23.6.0",
    "prettier": "^1.14.3",
    "rimraf": "^2.6.2",
    "ts-jest": "^23.10.3",
    "ts-node": "^7.0.1",
    "tslint": "^5.11.0",
    "tslint-config-prettier": "^1.15.0",
    "typescript": "^3.1.1"
  },
  "engines": {
    "node": ">=10.0.0"
  },
  "jest": {
    "preset": "ts-jest"
  },
  "gitHead": "403eb012f8b34bc115dd7dd86f80c76b39329f3b"
}
