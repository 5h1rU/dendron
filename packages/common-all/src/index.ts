export * from "./types";
export * from "./store";
export * from "./node";
export * from "./helpers";
