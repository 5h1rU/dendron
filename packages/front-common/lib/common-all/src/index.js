export * from "./node";
export * from "./helpers";
export * from "./integrations/dropbox";
//# sourceMappingURL=index.js.map