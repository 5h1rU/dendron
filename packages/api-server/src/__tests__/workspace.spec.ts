import { DendronAPI } from "@dendronhq/common-all";
import {
  EngineTestUtils,
  NodeTestUtils,
  tmpDir,
} from "@dendronhq/common-server";
import fs from "fs-extra";
import path from "path";

describe("main", () => {
  let wsRoot: string;
  let vault: string;

  beforeEach(async () => {
    wsRoot = tmpDir().name;
    vault = path.join(wsRoot, "vault");
    fs.ensureDirSync(vault);
    await EngineTestUtils.setupVault({
      vaultDir: vault,
      initDirCb: (dirPath: string) => {
        NodeTestUtils.createNotes(dirPath, [
          {
            id: "id.foo",
            fname: "foo",
          },
        ]);
      },
    });
  });

  test("one", async () => {
    const payload = {
      uri: wsRoot,
      config: {
        vaults: [vault],
      },
    };
    const api = new DendronAPI({
      endpoint: "http://localhost:3005",
      apiPath: "api",
    });
    await api.workspaceInit(payload);
    const resp = await api.workspaceList();
    expect(resp).toMatchSnapshot();
  });
});
