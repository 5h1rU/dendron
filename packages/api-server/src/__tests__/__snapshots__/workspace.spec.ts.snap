// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main one 1`] = `
Object {
  "data": Object {
    "notes": Array [
      Object {
        "body": "
",
        "children": Array [
          "id.foo",
        ],
        "created": 1601400963862,
        "custom": Object {},
        "data": Object {},
        "desc": "",
        "fname": "root",
        "id": "root",
        "parent": null,
        "stub": false,
        "title": "root",
        "type": "note",
        "updated": 1601400963862,
      },
      Object {
        "body": "foo body
",
        "children": Array [],
        "created": 1601400963868,
        "custom": Object {},
        "data": Object {},
        "desc": "",
        "fname": "foo",
        "id": "id.foo",
        "parent": "root",
        "stub": false,
        "title": "Foo",
        "type": "note",
        "updated": 1601400963868,
      },
    ],
    "schemas": Array [
      Object {
        "body": "",
        "children": Array [],
        "created": 1601400963897,
        "custom": Object {},
        "data": Object {},
        "desc": "",
        "fname": "root.schema",
        "id": "root",
        "parent": null,
        "stub": false,
        "title": "root",
        "type": "schema",
        "updated": 1601400963897,
      },
    ],
  },
  "error": null,
}
`;

exports[`main one 2`] = `
Object {
  "data": Object {
    "data": Object {
      "workspaces": Array [
        "ws:/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-26122-m67o1Aki0lIG",
        "ws:/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-27091-7nRusabb1Enc",
        "ws:/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-27536-M6FOp1Plp2D7",
      ],
    },
    "error": null,
  },
  "error": null,
}
`;
